stix_shifter_modules/gcp_chronicle/stix_transmission/api_client.py

```diff
@@ -149,7 +149,12 @@ def get_search_results(self, search_id, next_page_token, page_size):
 
         list_detection_endpoint = self.host + "/" + list_detection
 
-        return self.http_client.request(list_detection_endpoint, 'GET')
+        response = self.http_client.request(list_detection_endpoint, 'GET')
+
+        while response[0].status == 429:  # add sleep of 1 sec when resource exhaustion happens
+            time.sleep(1)
```

Comment:
are you sure sleeping 1 second resolve the resource exhaustion? I don't think you need to sleep if api returns 429. if resource is exhausted then just return whatever response you have got from the api.